sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/backup_key.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Defines the backupKey type.
+ *
+ */
+
+#pragma once
+
+#include "azure/keyvault/keys/dll_import_export.hpp"
+
+#include <cstdint>
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  /**
+   * @brief The JsonWebKey types.
+   *
+   */
+  class BackupKey final {
+  private:
+    std::vector<uint8_t> m_value;
+
+    BackupKey() = delete;
+
+  public:
+    /**
+     * @brief Construct a new backupKey type object.
+     *
+     * @param backupKey The backup key array data.
+     */
+    explicit BackupKey(std::vector<uint8_t> backupKey) : m_value(std::move(backupKey)) {}
+
+    /**
+     * @brief Return the backup key.
+     *
+     */
+    std::vector<uint8_t> const& GetValue() const { return m_value; }
+  };
```

Comment:
@heaths and @tg-msft we decided to create a "model type" wrapping the `std::vector<uint8_t>` return type from the `BackupKey` operation to future proof that API in case the service adds more things to it.

https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs#L883

We believe this is a good change following the C++ and general SDK guidelines around return types, since we can't overload based on that, in the future, and we value language specific consistency over consistency across languages. If you have any thoughts, on the contrary, please let me know :)

Similar to:
https://github.com/Azure/azure-sdk-for-net/blob/7e0315a4c606a504088476d00e335490e72c6fcc/sdk/keyvault/Microsoft.Azure.KeyVault/src/Generated/Models/BackupKeyResult.cs#L21

Essentially, we are leveraging the benefit of hindsight that we happen to have in C++.